src/components/sidenav/sidenav.spec.js

```diff
@@ -173,6 +173,85 @@ describe('mdSidenav', function() {
 
   });
 
+  describe("focus", function() {
+
+    var $material, $mdInteraction;
+
+    beforeEach( inject(function(_$material_, _$mdInteraction_) {
+      $material = _$material_;
+      $mdInteraction = _$mdInteraction_;
+    }));
+
+    function flush() {
+      $material.flushInterimElement();
+    }
+
+    function setupTrigger() {
+      var el;
+      inject(function($compile, $rootScope) {
+        var parent = angular.element(document.body);
+        el = angular.element('<button aria-label="Toggle Menu">Toggle</button>');
+        parent.append(el);
+        $compile(parent)($rootScope);
+        $rootScope.$apply();
+      });
+      return el;
+    }
+
+    it("should restore after sidenav triggered by keyboard", function(done) {
+      var sidenavElement = setup('');
+      var triggerElement = setupTrigger();
+      var controller = sidenavElement.controller('mdSidenav');
+
+      triggerElement.focus();
+
+      var evt = document.createEvent("KeyboardEvent");
+      evt.initEvent("keydown", true, true, window, 0, 0, 0, 0, 13, 13);
+      triggerElement[0].dispatchEvent(evt);
+
+      controller.$toggleOpen(true);
+      flush();
+
+      triggerElement.blur();
+
+      expect(document.activeElement).not.toBe(triggerElement[0]);
```

Comment:
This test is focusing the triggering element, then it executes a click to notify the ```ìnteraction``` service, then I open the Sidenav (all things are setup, like a real keydown). Then I manually open the sidenav with ```$toggleOpen````. After its opened I set the focus to the body (because PhantomJS isn't setting the focus as preferred to the sidenav). Then I validate the activeElement not to be the triggeringElement (because the Sidenav / Body is focused). After that I close that sidenav manually, and the sidenav thoughts I triggered the sidenav with keyboard, so its restoring my old activeElement.